shell.cpp

```diff
@@ -0,0 +1,48 @@
+#include<cstdio>
+/*Vick Ye
+ *38m 06s
+ *10/10
+ *shell numbers into indices
+ */
+int position[3] = {1, 2, 3};
+int A[100], B[100], G[100];
+
+void swap(int i, int j) {
+	int temp = position[i];
+	position[i] = position[j];
+	position[j] = temp;
+}
+
+int main() {
+	FILE * in = fopen("shell.in", "r");
+	int n;
+	fscanf(in, "%d", &n);
+	for(int i = 0; i < n; i++) {
+		fscanf(in, "%d %d %d", &A[i], &B[i], &G[i]);
+	}
+	fclose(in);
+	
```

Comment:
Add a *pre-processing* step:
```
for i=0:n
    A[I]--;
    B[I]--;
    G[I]--;
```
Take care of the 1-base vs 0-base conversion at the beginning and end. In the algorithm, it is always 0-based.
```